lib/TPP/Conversion/ConvertLinalgToXsmm/ConvertLinalgToXsmm.cpp

```diff
@@ -1085,7 +1109,13 @@ struct ConvertGenericToVnniMatmulLikeOp
     int64_t kPos = 1;
     if (hasBatch)
       kPos++;
-    int64_t k = cast<ShapedType>(bufferA.getType()).getShape()[kPos];
+    // Take the whole reduction dim size. Account for the VNNI factor (ensured
+    // by the earlier check) that splits the K dim in the shape.
+    std::optional<int64_t> vnniFactor =
+        vnni::utils::getVnniBlockingFactor(bufferB.getType());
+    assert(vnniFactor && "Must be in VNNI format");
```

Comment:
assert only works in debug mode, this can still crash in release mode

Reply:
Fair, added proper check.